File: JYRadarChartDemo/ViewController.m
//
//  ViewController.m
//  JYRadarChartDemo
//
//  Created by jy on 13-10-31.
//  Copyright (c) 2013年 wcode. All rights reserved.
//

#import "ViewController.h"
#import "JYRadarChart.h"

@interface ViewController () {
	JYRadarChart *p;
	JYRadarChart *p2;
}

@end

@implementation ViewController

- (void)viewDidLoad {
	[super viewDidLoad];

	p = [[JYRadarChart alloc] initWithFrame:CGRectMake(30, 20, 200, 200)];

	NSArray *a1 = @[@(81), @(97), @(87), @(60), @(65), @(77)];
	NSArray *a2 = @[@(91), @(87), @(33), @(77), @(78), @(96)];
	p.dataSeries = @[a1, a2];
	p.steps = 1;
	p.showStepText = YES;
	p.backgroundColor = [UIColor whiteColor];
	p.r = 60;
	p.minValue = 20;
	p.maxValue = 120;
	p.fillArea = YES;
	p.colorOpacity = 0.7;
    p.backgroundFillColor = [UIColor colorWithRed:0.9 green:0.9 blue:0.9 alpha:1];
	p.attributes = @[@"Attack", @"Defense", @"Speed", @"HP", @"MP", @"IQ"];
	p.showLegend = YES;
	[p setTitles:@[@"archer", @"footman"]];
	[p setColors:@[[UIColor yellowColor], [UIColor purpleColor]]];
	[self.view addSubview:p];


	[NSTimer scheduledTimerWithTimeInterval:1.5 target:self selector:@selector(updateData) userInfo:nil repeats:YES];



	p2 = [[JYRadarChart alloc] initWithFrame:CGRectMake(10, 220, 280, 200)];
	p2.centerPoint = CGPointMake(130, 100);
	p2.showLegend = YES;
    p2.backgroundFillColor = [UIColor whiteColor];
	[p2 setTitles:@[@"a", @"b", @"c", @"d", @"e", @"f", @"g", @"h", @"i", @"j"]];
	p2.drawPoints = YES;
	p2.attributes = @[@"Price", @"Value", @"Pressure", @"Height", @"Weight", @"Grade",
	                  @"Volume", @"Length", @"Size", @"Padding", @"Pages", @"HAHAHA"];
	NSArray *b1 = @[@(61), @(97), @(87), @(60), @(85), @(77), @(73), @(74), @(53), @(82), @(65), @(61)];
	NSArray *b2 = @[@(91), @(87), @(43), @(77), @(78), @(96), @(51), @(65), @(77), @(55), @(84), @(91)];
	NSArray *b3 = @[@(51), @(97), @(87), @(60), @(25), @(77), @(93), @(14), @(53), @(34), @(65), @(51)];
	NSArray *b4 = @[@(11), @(87), @(65), @(77), @(55), @(84), @(43), @(77), @(78), @(96), @(51), @(11)];
	NSArray *b5 = @[@(41), @(97), @(87), @(60), @(95), @(77), @(73), @(74), @(59), @(82), @(95), @(41)];
	NSArray *b6 = @[@(61), @(96), @(51), @(65), @(77), @(87), @(43), @(70), @(78), @(55), @(44), @(61)];
	NSArray *b7 = @[@(81), @(97), @(74), @(53), @(82), @(65), @(87), @(60), @(85), @(77), @(73), @(81)];
	NSArray *b8 = @[@(91), @(84), @(43), @(67), @(78), @(96), @(47), @(55), @(67), @(55), @(51), @(91)];
	NSArray *b9 = @[@(38), @(85), @(77), @(93), @(74), @(53), @(82), @(97), @(87), @(60), @(65), @(38)];
	NSArray *b10 = @[@(31), @(87), @(43), @(37), @(78), @(96), @(51), @(65), @(17), @(55), @(54), @(31)];
	p2.dataSeries = @[b1, b2, b3, b4, b5, b6, b7, b8, b9, b10];
	p2.steps = 2;
	p2.backgroundColor = [UIColor grayColor];
	[self.view addSubview:p2];
}

- (void)updateData {
	int n = 7;
	NSMutableArray *a = [NSMutableArray array];
	NSMutableArray *b = [NSMutableArray array];
	NSMutableArray *c = [NSMutableArray array];


	for (int i = 0; i < n - 1; i++) {
		a[i] = [NSNumber numberWithInt:arc4random() % 40 + 80];
		b[i] = [NSNumber numberWithInt:arc4random() % 50 + 70];
		c[i] = [NSNumber numberWithInt:arc4random() % 60 + 60];
	}

	p.dataSeries = @[a, b, c];
	p.steps = arc4random() % 6;
	p.fillArea = arc4random() % 2 ? YES : NO;
	p.drawPoints = arc4random() % 2 ? YES : NO;
	p.showStepText = arc4random() % 2 ? YES : NO;
	[p setTitles:@[@"iPhone", @"pizza", @"hard drive"]];
	[p setNeedsDisplay];
}

- (void)didReceiveMemoryWarning {
	[super didReceiveMemoryWarning];
	// Dispose of any resources that can be recreated.
}

@end

File: JYRadarChartSwiftDemo/JYRadarChartSwiftDemo/ViewController.swift
//
//  ViewController.swift
//  JYRadarChartSwiftDemo
//
//  Created by Erick Santos on 30/04/17.
//
//

import UIKit

class ViewController: UIViewController {

    let p = JYRadarChart(frame: CGRect(x: 30.0, y: 20.0, width: 200.0, height: 200.0))
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let a1: Array<Double> = [81.0, 97.0, 87.0, 60.0, 65.0, 77.0]
        let a2: Array<Double> = [91.0, 87.0, 33.0, 77.0, 78.0, 96.0]
        p.data = [a1, a2]
        p.steps = 1
        p.isShowStepText = true
        p.backgroundColor = UIColor.white
        p.r = 60
        p.minValue = 20
        p.maxValue = 120
        p.isFillArea = true
        p.colorOpacity = 0.7
        p.backgroundFillColor = UIColor(red: 0.9, green: 0.9, blue: 0.9, alpha: 1.0)
        p.attributes = ["Attack", "Defense", "Speed", "HP", "MP", "IQ"]
        p.isShowLegend = true
        p.titles = ["archer", "footman"]
        p.colors = [UIColor.yellow, UIColor.purple];
        p.isShowTotal = false
        self.view.addSubview(p)
        
        Timer.scheduledTimer(timeInterval: 1.5, target: self, selector: #selector(updateData), userInfo: nil, repeats: true)
        
        
        let p2 = JYRadarChart(frame: CGRect(x: 10.0, y: 220.0, width: 280.0, height: 200.0))
        p2.centerPoint = CGPoint(x: 130, y: 100)
        p2.isShowLegend = true
        p2.backgroundFillColor = UIColor.white
        p2.titles = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]
        p2.isDrawPoints = true
        p2.attributes = ["Price", "Value", "Pressure", "Height", "Weight", "Grade", "Volume", "Length", "Size", "Padding", "Pages", "HAHAHA"]
        let b1: Array<Double> = [61.0, 97.0, 87.0, 60.0, 85.0, 77.0, 73.0, 74.0, 53.0, 82.0, 65.0, 61.0]
        let b2: Array<Double> = [91.0, 87.0, 43.0, 77.0, 78.0, 96.0, 51.0, 65.0, 77.0, 55.0, 84.0, 91.0]
        let b3: Array<Double> = [51.0, 97.0, 87.0, 60.0, 25.0, 77.0, 93.0, 14.0, 53.0, 34.0, 65.0, 51.0]
        let b4: Array<Double> = [11.0, 87.0, 65.0, 77.0, 55.0, 84.0, 43.0, 77.0, 78.0, 96.0, 51.0, 11.0]
        let b5: Array<Double> = [41.0, 97.0, 87.0, 60.0, 95.0, 77.0, 73.0, 74.0, 59.0, 82.0, 95.0, 41.0]
        let b6: Array<Double> = [61.0, 96.0, 51.0, 65.0, 77.0, 87.0, 43.0, 70.0, 78.0, 55.0, 44.0, 61.0]
        let b7: Array<Double> = [81.0, 97.0, 74.0, 53.0, 82.0, 65.0, 87.0, 60.0, 85.0, 77.0, 73.0, 81.0]
        let b8: Array<Double> = [91.0, 84.0, 43.0, 67.0, 78.0, 96.0, 47.0, 55.0, 67.0, 55.0, 51.0, 91.0]
        let b9: Array<Double> = [38.0, 85.0, 77.0, 93.0, 74.0, 53.0, 82.0, 97.0, 87.0, 60.0, 65.0, 38.0]
        let b10: Array<Double> = [31.0, 87.0, 43.0, 37.0, 78.0, 96.0, 51.0, 65.0, 17.0, 55.0, 54.0, 31.0]
        p2.data = [b1, b2, b3, b4, b5, b6, b7, b8, b9, b10]
        p2.steps = 2;
        p2.isShowTotal = false
        p2.backgroundColor = UIColor.gray
        self.view.addSubview(p2)
    }
    
    func updateData() {
        let n = 6
        var a = Array<Double>()
        var b = Array<Double>()
        var c = Array<Double>()
        
        
        for _ in 0..<n {
            a.append(Double(arc4random() % 40 + 80))
            b.append(Double(arc4random() % 50 + 70))
            c.append(Double(arc4random() % 60 + 60))
        }
        
        p.data = [a, b, c]
        p.steps = Int(arc4random() % 6) + 1
        p.isFillArea = arc4random() % 2 == 0 ? true : false
        p.isDrawPoints = arc4random() % 2 == 0 ? true : false
        p.isShowStepText = arc4random() % 2 == 0 ? true : false
        p.titles = ["iPhone", "pizza", "hard drive"]
        p.setNeedsDisplay()
    }

}

